### DISCUSSION TITLE
How did Patrick know to write this line the way he did? (Early in Lesson 3-creating list of contracts)

### DISCUSSION THREAD

**Discussion by DeanKrupka** [Cyfrin/foundry-full-course-cu#1047] [Q&A] [answered]
My specific question refers to the first line of code within the function below:
'''
        SimpleStorage newSimpleStorageContract = new SimpleStorage();
'''

Im new to coding, so I know this is basic, but what is the rationale for the very first type "SimpleStorage" in that line of code? Full code below:

//SPDX-License-Identifier: MIT
pragma solidity ^0.8.19;

import {SimpleStorage} from "./SimpleStorage.sol"; // named imports are more targeted

contract StorageFactory {
    SimpleStorage[] public listOfSimpleStorageContracts; // creates variable for full simple storage contract

    function createSimpleStorageContract() public {
        SimpleStorage newSimpleStorageContract = new SimpleStorage();
        listOfSimpleStorageContracts.push(newSimpleStorageContract);

    }
}

**Comment by Packagefather** [comment, marked as answer]
The function you are talking about is deploying an instance of SimpleStorage contract. 


Your newly deployed is taking up the SimpleStorage data type which you stated 
``` SimpleStorage [] public ; ```



Also, you know how you declare variables, eg. uint256  amount, address user, bool success, that's what you are doing there, you are saying that the new contract is of type SimpleStorage which you imported it's interface earlier with this 
```import {SimpleStorage} from "./SimpleStorage.sol"; ```

Hope that helps more than confuse, still learning to best explain things.

**Reply by DeanKrupka** [reply]
Thanks very much. Definitely helps.
